### PR TITLE
Fixes linting issues

### PR DESCRIPTION
**Pull request by dukeofsussex**
Fixes #

- [x] Unit tests created and pass
- [x] JS Docs have been updated

**Description:**
Start fixing those bithound complaints (going to have to be done sooner or later)

@pencilblue/owners

@brianhyder Please check the second commit, needs thorough reviewing.

### CODE REVIEW

**Comment by dukeofsussex** [comment]
Bithound not complaining? How odd

File: include/service/media/renderers/daily_motion_media_renderer.js [hunk 1]
```diff
@@ -149,7 +149,7 @@ module.exports = function AudioMediaRendererModule(pb) {
      * @return {Boolean} TRUE if URL points to the main domain and media resource, FALSE if not
      */
     DailyMotionMediaRenderer.isLibyanDomain = function(parsedUrl) {
-        if (util.isString(parsedUrl)) {
+        if (util.isString(parsedUrl)) { // Necessary?
```
Comment (brianhyder) [inline include/service/media/renderers/daily_motion_media_renderer.js]: if I recall correctly these were necessary based on some code paths.  Please review the comment and the other for this file.

File: include/service/json_fs_entity_service.js [hunk 2]
```diff
@@ -74,7 +74,7 @@ module.exports = function JSONFSEntityServiceModule(pb) {
      */
     JSONFSEntityService.prototype.set = function(key, value, cb) {
         if (!util.isObject(value) && !util.isArray(value)) {
-            cb(new PBError(this.type+": Value must be an array or object: "+util.inspect(value)), null);
+            cb(new pb.PBError(this.type+": Value must be an array or object: "+util.inspect(value)), null); // PBError class necessary?
```
Comment (brianhyder) [inline include/service/json_fs_entity_service.js]: I do not believe it is.  At one point, early on, there was special logic to handle these errors.  That is no longer the case.  However, without tests there is not way to no for sure.  I would just leave the comments in for now.

File: include/service/media/renderers/instagram_media_renderer.js [hunk 3]
```diff
@@ -129,7 +129,7 @@ module.exports = function InstagramMediaRendererModule(pb) {
      * @return {Boolean} TRUE if URL points to the main domain and media resource, FALSE if not
      */
     InstagramMediaRenderer.isFullSite = function(parsedUrl) {
-        if (util.isString(parsedUrl)) {
+        if (util.isString(parsedUrl)) { // Necessary?
```
Comment (brianhyder) [inline include/service/media/renderers/instagram_media_renderer.js]: please remove the comment

File: include/service/media/renderers/kickstarter_media_renderer.js [hunk 4]
```diff
@@ -129,7 +129,7 @@ module.exports = function KickStarterMediaRendererModule(pb) {
      * @return {Boolean} TRUE if URL points to the main domain and media resource, FALSE if not
      */
     KickStarterMediaRenderer.isFullSite = function(parsedUrl) {
-        if (util.isString(parsedUrl)) {
+        if (util.isString(parsedUrl)) { // Necessary?
```
Comment (brianhyder) [inline include/service/media/renderers/kickstarter_media_renderer.js]: please remove the comment

File: include/service/media/renderers/daily_motion_media_renderer.js (hunk 1 above)
Comment (dukeofsussex) [inline include/service/media/renderers/daily_motion_media_renderer.js]: What has me confused is that a parsedUrl is being parsed again

**Comment by dukeofsussex** [comment]
@brianhyder I was going to remove all comments, most are just there to clarify stuff. And all the "Necessary?" ones are duplicates, as the corresponding code snippet has been copied to every renderer and semantically doesn't make sense

File: plugins/pencilblue/controllers/admin/content/articles/manage_articles.js [hunk 5]
```diff
@@ -34,7 +34,7 @@ module.exports = function(pb) {
     ManageArticles.prototype.render = function(cb) {
         var self = this;
         if(!pb.security.isAuthorized(this.session, {logged_in: true, admin_level: pb.SecurityService.ACCESS_EDITOR})) {
-            where.author = this.session.authentication.user_id;
+            where.author = this.session.authentication.user_id; // Where is the object? Is this just a copy-paste?
```
Comment (brianhyder) [inline plugins/pencilblue/controllers/admin/content/articles/manage_articles.js]: This is set when the user logs in

Comment (dukeofsussex) [inline plugins/pencilblue/controllers/admin/content/articles/manage_articles.js]: This one actually gave me an error today, stating `where is not defined` when viewing the exact page as a writer

File: plugins/portfolio/controllers/blog.js [hunk 6]
```diff
@@ -394,7 +394,7 @@ module.exports = function BlogModule(pb) {
         if(this.req.pencilblue_article || this.req.pencilblue_page) {
             cb(null, content.headline + ' | ' + pb.config.siteName);
         }
-        else if(searchId = this.req.pencilblue_section || this.req.pencilblue_topic) {
+        else if(searchId = this.req.pencilblue_section || this.req.pencilblue_topic) { // Is this an assignment during evaluation?
```
Comment (brianhyder) [inline plugins/portfolio/controllers/blog.js]: yes, the order of operations coalesces the section and topic before assigning them to the search id variable

**Comment by brianhyder** [comment]
For the most part this looks great.  Just need to remove a few inline comments and then i'll merge.

**Comment by dukeofsussex** [comment]
I'll tidy it up and let you know when it's ready

**Comment by dukeofsussex** [comment]
@brianhyder Is include/service/entities/theme_service.js used anywhere? Couldn't find anything

File: plugins/pencilblue/controllers/admin/content/articles/manage_articles.js (hunk 5 above)
Comment (brianhyder) [inline plugins/pencilblue/controllers/admin/content/articles/manage_articles.js]: Oh, I see.  This is related to #980.

File: include/service/media/renderers/daily_motion_media_renderer.js (hunk 1 above)
Comment (dukeofsussex) [inline include/service/media/renderers/daily_motion_media_renderer.js]: Okay, I'll leave it be. Just seems redundant, but it's only a small check.

**Comment by brianhyder** [comment]
The theme_service.js file may not be.  There are a few other unused files too but we are waiting until the next version to rip it out so that we don't break anything that other users were relying on.

File: plugins/pencilblue/controllers/admin/content/articles/manage_articles.js (hunk 5 above)
Comment (dukeofsussex) [inline plugins/pencilblue/controllers/admin/content/articles/manage_articles.js]: Ah right. Just going to leave it then, too bad bithound :smile:

**Comment by dukeofsussex** [comment]
Should be good to go, already rebased onto 0.8.0 for an easy merge (I hope 😉 )

File: plugins/pencilblue/controllers/preview.js [hunk 7]
```diff
@@ -56,10 +55,6 @@ module.exports = function PreviewModule(pb) {
         });
     };
 
-    Preview.prototype.getPageTitle = function() {
-        return article.name;
-    };
```
Comment (dukeofsussex) [inline plugins/pencilblue/controllers/preview.js]: The case was redundant,  as it uses the same as the default case, so why have it? (and bithound didn't like the syntax for whatever reason) And if I'm not mistaken this function isn't

**Pull request merged**